code/modules/antagonists/heretic/status_effects/buffs.dm

```diff
@@ -58,12 +75,24 @@
 
 /datum/status_effect/marshal/on_remove()
 	owner.remove_movespeed_mod_immunities(id, /datum/movespeed_modifier/damage_slowdown)
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/drinker = owner
+	for(var/obj/item/bodypart/potentially_wounded in drinker.bodyparts)
+		for(var/datum/wound/found_wound in potentially_wounded.wounds)
+			found_wound.remove_wound()
+	if(length(drinker.get_missing_limbs()) >= 1)
```

Comment:
```suggestion
	if(length(drinker.get_missing_limbs()))
```
The if statements won't pass if length() returns zero anyway, and it cannot return anything below that.